### PR TITLE
OS-206 add ssh and environment controls

### PR DESCRIPTION
**Pull request by alkazar**
**Why**:
Currently, Grid handles changing log level, the environment (red/black/prod), and enabling/disabling ssh. It does this by using sudo and a password.

Adding this functionality to hwctl allows Grid to call the script without a password or using sudo since hwctl can already be called with elevated privileges using pkexec.

**How**:
 - Added an `ssh` subsystem with `enable` and `disable` commands which call `systemctl`.
 - Added an `environment` subsystem with `set` and `clear` commands which modify config files under `~/.config/environment.d` which is read by the gamescope session at startup.
 - The `environment` subsystem allows setting any environment variable by creating a separate config file based on the environment variable name to prevent the variables overwriting each other, security was considered and addressed by the following measures:
   - The `environment` subsystem cannot be called by root and can only operate on user files
   - The environment variable name that is passed in is validated to prevent someone passing relative path names to trick the script into operating on files in the wrong location.
   - NOTE: `XDG_CONFIG_HOME` is honored, which could be abused, but in that case the whole system is likely compromised and the threat is mitigated by disallowing operation as the root user.

**Testing**:
 - All functionality was tested on my local development machine

### CODE REVIEW

File: usr/bin/hwctl [hunk 1]
```diff
@@ -495,12 +495,98 @@ function datetime {
...
 
 SUBSYSTEM=$1
 shift
 
-if [[ "$SUBSYSTEM" == @(audio|datetime|display|system-battery|system-info|storage) ]]; then
+if [[ "$SUBSYSTEM" == @(audio|datetime|display|system-battery|system-info|storage|ssh|environment) ]]; then
 	$SUBSYSTEM $@
 else
 	echo "Error: unknown subsystem $SUBSYSTEM"
+	echo "Available subsystems:"
```
Comment (alkazar) [inline usr/bin/hwctl:583]: Mostly unrelated change to help with usage.

**Review by Ericky14 — APPROVED** [review]

**Review by LukeShortCloud — APPROVED** [review]
Working good on my end, too. Thanks!

```
$ sudo ./hwctl ssh
Error: unknown command
$ sudo ./hwctl ssh enable
Created symlink /etc/systemd/system/multi-user.target.wants/sshd.service → /usr/lib/systemd/system/sshd.service.
$ sudo ./hwctl ssh disable
Removed "/etc/systemd/system/multi-user.target.wants/sshd.service".
```

```
$ ./hwctl environment
Error: unknown command
$ ./hwctl environment set
Error: No environment variable specified
$ ./hwctl environment set foobar
Error: No value specified
$ ./hwctl environment set foobar 123
$ cat ~/.config/environment.d/foobar.conf 
foobar=123
$ ./hwctl environment clear foobar
$ cat ~/.config/environment.d/foobar.conf 
cat: /home/user/.config/environment.d/foobar.conf: No such file or directory
```
